prober/icmp.go

```diff
@@ -243,13 +279,53 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 
 	rb := make([]byte, 65536)
 	deadline, _ := ctx.Deadline()
-	if err := socket.SetReadDeadline(deadline); err != nil {
+	if icmpConn != nil {
+		err = icmpConn.SetReadDeadline(deadline)
+	} else {
+		err = v4RawConn.SetReadDeadline(deadline)
+	}
+	if err != nil {
 		level.Error(logger).Log("msg", "Error setting socket deadline", "err", err)
 		return
 	}
 	level.Info(logger).Log("msg", "Waiting for reply packets")
 	for {
-		n, peer, err := socket.ReadFrom(rb)
+		var n int
+		var peer net.Addr
+		var err error
+		var hopLimit float64 = -1
+
+		if dstIPAddr.IP.To4() == nil {
+			var cm *ipv6.ControlMessage
+			n, cm, peer, err = icmpConn.IPv6PacketConn().ReadFrom(rb)
+			// HopLimit == 0 is valid for IPv6, although go initialize it as 0
+			if cm != nil && ipv6HopLimitFlagSet {
+				hopLimit = float64(cm.HopLimit)
+			} else {
+				level.Debug(logger).Log("msg", "Cannot get Hop Limit from the received packet. 'probe_icmp_reply_hop_limit' will be missing.")
+			}
+		} else {
+			var cm *ipv4.ControlMessage
+			if icmpConn != nil {
+				n, cm, peer, err = icmpConn.IPv4PacketConn().ReadFrom(rb)
+			} else {
+				var h *ipv4.Header
+				var p []byte
+				h, p, cm, err = v4RawConn.ReadFrom(rb)
+				if err == nil {
+					copy(rb, p)
+					n = len(p)
+					peer = &net.IPAddr{IP: h.Src}
+				}
+			}
+			// TTL == 0 is not valid for IPv4
+			if cm != nil && cm.TTL > 0 {
+				/* Not really hopLimit, but it is in practice */
```

Comment:
Be consistent in comment style, and all comments should end with a full stop.

Reply:
ok. I also switched from /*x*/ to //x